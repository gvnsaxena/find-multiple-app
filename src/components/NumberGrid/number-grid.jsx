import React,  {memo} from 'react';
import './styles/number-grid.scss';

const NumberGrid = ({startCount, endCount, 
            clickHandler, findMultiplesOfNumbers}) => {

    const makeArray = () => {
        let arr = [];
        for(let i=startCount; i<=endCount; i++){
            arr.push(i);
        }
        return arr;
    }

    return(
        <div className='number-grid'>
            {makeArray().map((item) => 
                <div onClick={clickHandler} key={item} aria-label={`Click Number ${item}`}
                    className={`number-box ${findMultiplesOfNumbers(item) ? 'highlight-number': ''}`}>{item}
                </div>
            )}
        </div>
    );
}

export default memo(NumberGrid);