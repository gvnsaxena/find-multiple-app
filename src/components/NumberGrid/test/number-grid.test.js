import { screen, render } from '@testing-library/react';
import NumberGrid from '../number-grid';

it('should render and match the snapshot', () => {
  const {
    container: { firstChild },
  } = render(<NumberGrid startCount='1' endCount='10'
    clickHandler={jest.fn()} findMultiplesOfNumbers={jest.fn().mockImplementation(()=> true)}/>);
  expect(firstChild).toMatchSnapshot();
});

it('should render Number grid with 10 box, aria-label and highlighted Class', () => {
    render(<NumberGrid startCount='1' endCount='10'
      clickHandler={jest.fn()} findMultiplesOfNumbers={jest.fn().mockImplementation(()=> true)}/>);
      expect(screen.getByText(10).textContent).toEqual('10');
      expect(screen.getByText(10).getAttribute('aria-label')).toEqual('Click Number 10');
      expect(screen.getByText(10).className).toEqual('number-box highlight-number');
  });

  it('should render Number grid with 130 box, aria-label and without Highlighted class', () => {
    render(<NumberGrid startCount='1' endCount='130'
      clickHandler={jest.fn()} findMultiplesOfNumbers={jest.fn().mockImplementation(()=> false)}/>);
      expect(screen.getByText(130).textContent).toEqual('130');
      expect(screen.getByText(130).getAttribute('aria-label')).toEqual('Click Number 130');
      expect(screen.getByText(130).className).toEqual('number-box ');
  });