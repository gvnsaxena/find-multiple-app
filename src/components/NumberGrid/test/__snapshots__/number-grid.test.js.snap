// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render and match the snapshot 1`] = `
<div
  class="number-grid"
>
  <div
    aria-label="Click Number 1"
    class="number-box highlight-number"
  >
    1
  </div>
  <div
    aria-label="Click Number 2"
    class="number-box highlight-number"
  >
    2
  </div>
  <div
    aria-label="Click Number 3"
    class="number-box highlight-number"
  >
    3
  </div>
  <div
    aria-label="Click Number 4"
    class="number-box highlight-number"
  >
    4
  </div>
  <div
    aria-label="Click Number 5"
    class="number-box highlight-number"
  >
    5
  </div>
  <div
    aria-label="Click Number 6"
    class="number-box highlight-number"
  >
    6
  </div>
  <div
    aria-label="Click Number 7"
    class="number-box highlight-number"
  >
    7
  </div>
  <div
    aria-label="Click Number 8"
    class="number-box highlight-number"
  >
    8
  </div>
  <div
    aria-label="Click Number 9"
    class="number-box highlight-number"
  >
    9
  </div>
  <div
    aria-label="Click Number 10"
    class="number-box highlight-number"
  >
    10
  </div>
</div>
`;
